backend/main.py

```
@@ -1080,8 +1129,20 @@ async def lifespan(app: FastAPI):
     )
+    # The watermark warm-up runs on its dedicated 1-worker pool; cancelling
+    # the task above detaches the asyncio side but a thread already inside
+    # the ~42s cold import keeps running (Python can't kill it — same
+    # reality as the GPU-pool note above). Drain the pool's QUEUE so nothing
+    # new starts, mirroring _reset_gpu_pool's bounded-abandon approach.
+    try:
+        from services.model_manager import get_watermark_pool as _get_wm_pool
+
+        _get_wm_pool().shutdown(wait=False, cancel_futures=True)
+    except Exception:
+        pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)